### PR TITLE
Move to internal

### PR DESCRIPTION
**Pull request by Opelord**
Last PR referencing issue KNTK-303
It builds on PR #60 

- Moved apiv6 contents (cloudexport, synthetics and localhost_apiserver) to root kentikapi directory.
- Renamed localhost_apiserver to localhost_api_fakeserver to better describe its purpose.
- Refactored code to reflect the changes.
- Moved code generating scripts to root.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+.idea
+test.log
```
Comment (Fenthick) [inline .gitignore:2]: Scripts in the repo do not produce test.log and lint.log files, so I don't see a reason for polluting a gitignore with it. I suggest just to ignore all log files (`*.log`) if it helps with dev process. Notice the alphabetical order.

File: README.md [hunk 2]
```diff
@@ -56,8 +56,8 @@ Additional development steps for API v6:
 - Generate documentation: `./generate_docs.sh`
 
 Note that due to the design of OpenAPI generator, some generated files of the stub server need to be filled manually. They are listed in following files:
-- [./apiv6/localhost_apiserver/cloudexport/.openapi-generator-ignore](./apiv6/localhost_apiserver/cloudexport/.openapi-generator-ignore)
-- [./apiv6/localhost_apiserver/synthetics/.openapi-generator-ignore](./apiv6/localhost_apiserver/synthetics/.openapi-generator-ignore)
+- [./apiv6/localhost_apiserver/cloudexport/.openapi-generator-ignore](kentikapi/localhost_apiserver/cloudexport/.openapi-generator-ignore)
```
Comment (Fenthick) [inline README.md:59]: Dead link

File: examples/cloud_export/main.go [hunk 3]
```diff
@@ -6,7 +6,8 @@ import (
 	"os"
 	"time"
 
-	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/cloudexport"
+	cloudexport2 "github.com/kentik/community_sdk_golang/kentikapi/cloudexport"
```
Comment (Fenthick) [inline examples/cloud_export/main.go:9]: Please remove cloudexport2 alias everywhere.

File: examples/synthetics/main.go [hunk 4]
```diff
@@ -6,7 +6,8 @@ import (
 	"os"
 	"time"
 
-	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/synthetics"
+	synthetics2 "github.com/kentik/community_sdk_golang/kentikapi/synthetics"
```
Comment (Fenthick) [inline examples/synthetics/main.go:9]: Please remove synthetics2 alias everywhere

File: kentikapi/client.go [hunk 5]
```diff
@@ -3,11 +3,11 @@ package kentikapi
 import (
```
Comment (Fenthick) [inline kentikapi/client.go:3]: generate_server.sh  generates the code to wrong directory. Please fix that.
Also, git apply command in the script fails because of bad path. If you fix the path it will probably still fails, because patch itself contains paths. You could try modifying the paths in patch by hand to fix that.

File: .golangci.yml [hunk 6]
```diff
@@ -4,8 +4,8 @@ run:
   build-tags:
     - examples
   skip-dirs:
-    - apiv6/localhost_apiserver/cloudexport
-    - apiv6/localhost_apiserver/synthetics
+    - localhost_api_fakeserver/cloudexport
```
Comment (Fenthick) [inline .golangci.yml:7]: When we were discussing renaming localhost_apiserver, I meant just _fakeserver_, not _localhost_api_fakeserver_. _localhost_ is unimportant implementation detail and we have _api_ already in _kentikapi_ parent directory, so I think they are redundant.
Please rename to _fakeserver_

File: README.md [hunk 7]
```diff
@@ -37,6 +40,26 @@ Run all tests, including usage examples: `go test -tags examples ./...`
...
+The OpenAPI Generator is used for code generation: <https://openapi-generator.tech/>.
+Generated code is checked-in to the repository, so that the user can _go get_ the library.
+
+Additional development steps for API v6:
+- Access apiv6 directory: cd apiv6
+- Generate the client: `./generate_client.sh`
+- Generate the stub server: `./generate_server.sh`
+- Generate documentation: `./generate_docs.sh`
+
+Note that due to the design of OpenAPI generator, some generated files of the stub server need to be filled manually. They are listed in following files:
```
Comment (mateuszmidor) [inline README.md:58]: Hmm this is not a "design of the the generator" problem, this is common across such generators that they generate only scaffolding that the programmer needs to fill with actual code, I would rephrase to:
_Note that the generated code scaffolding needs to be complemented with actual logic implementation_

File: examples/demos/cloud_export_demo/main.go [hunk 8]
```diff
@@ -4,9 +4,10 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/kentik/community_sdk_golang/apiv6/examples/demos"
-	"github.com/kentik/community_sdk_golang/apiv6/kentikapi"
-	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/cloudexport"
+	cloudexport2 "github.com/kentik/community_sdk_golang/kentikapi/cloudexport"
```
Comment (mateuszmidor) [inline examples/demos/cloud_export_demo/main.go:7]: Why is it called "caloudexport2"...?

File: examples/synthetics/mesh_test/main.go [hunk 9]
```diff
@@ -9,8 +9,9 @@ import (
 	"text/tabwriter"
 	"time"
 
-	"github.com/kentik/community_sdk_golang/apiv6/examples"
-	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/synthetics"
+	synthetics2 "github.com/kentik/community_sdk_golang/kentikapi/synthetics"
```
Comment (mateuszmidor) [inline examples/synthetics/mesh_test/main.go:12]: Why "synthetics2" ?

File: generate_server.sh [hunk 10]
```diff
@@ -54,7 +54,7 @@ function generate_cloud_export_server() {
 
 function generate_synthetics_server() {
     synthetics_package="syntheticsstub"
-    synthetics_server_output_dir="localhost_apiserver/synthetics"
+    synthetics_server_output_dir="localhost_fakeserver/synthetics"
```
Comment (mateuszmidor) [inline generate_server.sh:57]: Probably: ./kentikapi/localhost_fakeserver/synthetics

File: examples/cloud_export/main.go (hunk 3 above)
Comment (Fenthick) [reply examples/cloud_export/main.go]: Done

File: generate_server.sh [hunk 11]
```diff
@@ -45,7 +45,7 @@ function download_openapi_spec() {
 
 function generate_cloud_export_server() {
     cloud_export_package="cloudexportstub"
-    cloud_export_server_output_dir="localhost_apiserver/cloudexport"
+    cloud_export_server_output_dir="localhost_fakeserver/cloudexport"
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply generate_server.sh:48]: Definitely `./kentikapi/localhost_fakeserver/cloudexport`

File: .golangci.yml [hunk 12]
```diff
@@ -4,8 +4,8 @@ run:
   build-tags:
     - examples
   skip-dirs:
-    - apiv6/localhost_apiserver/cloudexport
-    - apiv6/localhost_apiserver/synthetics
+    - localhost_fakeserver/cloudexport
+    - localhost_fakeserver/synthetics
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply .golangci.yml:8]: I vote on fakeserver.
1. `localhost` is an implementation detail that is not the most important (IMO)
2. we have `api` already in kentikapi parent directory, so I think that can be omitted.
3. `fake` suggests that it is a test double of type `fake` (https://martinfowler.com/articles/mocksArentStubs.html). So it provides valuable information for package consumer.
4. "By convention, packages are given lower case, single-word names; there should be no need for underscores or mixedCaps." (https://golang.org/doc/effective_go#package-names).  `fakeserver` is short and closer to that convention than `localhost_apiserver`

File: README.md [hunk 13]
```diff
@@ -37,6 +40,26 @@ Run all tests, including usage examples: `go test -tags examples ./...`
...
+
+Subset of the code is generated from API specification available at:
+- <https://github.com/kentik/api-schema-public/tree/master/gen/openapiv2/kentik/cloud_export/v202101beta1>
+- <https://github.com/kentik/api-schema-public/tree/master/gen/openapiv2/kentik/synthetics/v202101beta1>
+
+The OpenAPI Generator is used for code generation: <https://openapi-generator.tech/>.
+Generated code is checked-in to the repository, so that the user can _go get_ the library.
+
+Additional development steps for API v6:
+- Access apiv6 directory: cd apiv6
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply README.md:53]: Yeah, those development steps should be merged now (L36)

File: .golangci.yml (hunk 12 above)
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply .golangci.yml:8]: Hmmmm...

_1.  localhost is an implementation detail that is not the most important (IMO)_
Well it has nothing to do with implementation; you can just as well run it remotely. The information provided here is that it is intended to be run locally, config, content and lifetime under user control
_2.  we have api already in kentikapi parent directory, so I think that can be omitted._
apiserver precisely tells what it is, there is no information duplication here. BTW shouldnt this server be in root dir? Why is it part of the client package....?
_3. fake suggests that it is a test double of type fake (https://martinfowler.com/articles/mocksArentStubs.html). So it provides valuable information for package consumer._
it should make no difference to the consumer if it is mock/stub/fake/double, and as a matter of fact, "fake" is implementation detail :)
_4. "By convention, packages are given lower case, single-word names; there should be no need for underscores or mixedCaps." (https://golang.org/doc/effective_go#package-names). fakeserver is short and closer to that convention than localhost_apiserver_
Well, there should neither be need for enumerators nor generics... ;)

I can eventually agree to "fakeserver", but I dont like it!

File: examples/demos/cloud_export_demo/main.go (hunk 8 above)
Comment (Opelord) [reply examples/demos/cloud_export_demo/main.go]: Since there were to imports ending in cloudexport GoLand must have created an alias for the added cloudexport. I didn't notice it at the time. Similarly with synthetics.

File: kentikapi/client.go (hunk 5 above)
Comment (Fenthick) [reply kentikapi/client.go:3]: Script fails with `error: can't open patch './kentikapi/fakeserver/patches/0001-Modify-fakeserver-not-to-require-preset-in-.patch': No such file or directory
`

**Review by Fenthick — CHANGES_REQUESTED** [review]

File: examples/synthetics/mesh_test/main.go [hunk 14]
```diff
@@ -9,8 +9,9 @@ import (
 	"text/tabwriter"
 	"time"
 
-	"github.com/kentik/community_sdk_golang/apiv6/examples"
-	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/synthetics"
+	synthetics "github.com/kentik/community_sdk_golang/kentikapi/synthetics"
```
Comment (Fenthick) [inline examples/synthetics/mesh_test/main.go:12]: Now this alias is redundant. Please just remove it

**Review by Fenthick — CHANGES_REQUESTED** [review]
